h/routes.py

```diff
@@ -181,6 +181,7 @@ def includeme(config):  # pylint: disable=too-many-statements
     # Client
     config.add_route("sidebar_app", "/app.html")
     config.add_route("notebook_app", "/notebook")
+    config.add_route("profile_app", "/user-profile")
```

Comment:
We discussed using `/profile` as the path. However, I have found this path exists in `h` since forever, and it is configured to redirect to `/account/settings`.

We should decide if this redirect is still relevant and remove it if it isn't, or just use a different path.

Reply:
> We should decide if this redirect is still relevant and remove it if it isn't, or just use a different path.

Checking the git history in h provides some insight into how likely it is that anyone is relying on the old views. The original `/profile` view was removed in a0d0eba3658da680ed1b446d3b246450061f50e0 (Oct 2015) with the comment:

> Remove standalone profile/edit_profile/disable_user views
> These views are broken, and I can't find any evidence that they ever actually worked.

On that basis I think we should feel free to use `/profile` as a URL for new functionality, though we might want to use a temporary URL while we develop the app. Something that occurred to me while looking at the new app is that I think of "profile" and "account settings" as different things in a social application:

- The user's profile is the identity they display to the outside world
- The user's settings includes things like contact methods (phone number, email) notifications, privacy, API keys etc. which affect their experience but not how others see them

We should discuss the routes and naming in more detail this week.

Reply:
> Checking the git history in h provides some insight into how likely it is that anyone is relying on the old views. The original /profile view was removed in https://github.com/hypothesis/h/commit/a0d0eba3658da680ed1b446d3b246450061f50e0 (Oct 2015) with the comment:

I'm thinking that we could even go a bit further, and check the logs to see if there's any request hitting this endpoint at all.

Reply:
> I'm thinking that we could even go a bit further, and check the logs to see if there's any request hitting this endpoint at all.

This redirect is served by a Pyramid "tweet" in h, which means that the redirect should definitely be visible in h's access logs, and I think redirects will be visible as transactions in New Relic, but I'm not certain. h's access logs I believe are available in Papertrail, but check with Ian on that.

Reply:
> This redirect is served by a Pyramid "tweet" in h, which means that the redirect should definitely be visible in h's access logs, and I think redirects will be visible as transactions in New Relic, but I'm not certain. h's access logs I believe are available in Papertrail, but check with Ian on that.

Here I swoop in pedantically, but I think, to ease confusion, it's important to point out that Rob means _tween_. GRIN.

Reply:
Yes, indeed. "tween" is short for "between" and IIRC it is similar to "middleware" in other frameworks.